cpg_workflows/inputs.py

```diff
@@ -8,41 +8,111 @@
 from cpg_workflows.filetypes import CramPath, GvcfPath
 
 from .metamist import AnalysisType, Assay, MetamistError, get_metamist, parse_reads
-from .targets import Cohort, PedigreeInfo, SequencingGroup, Sex
+from .targets import Cohort, MultiCohort, PedigreeInfo, SequencingGroup, Sex
 from .utils import exists
 
 _cohort: Cohort | None = None
+_multicohort: MultiCohort | None = None
 
 
-def get_cohort() -> Cohort:
+def actual_get_multicohort() -> MultiCohort:
+    """Return the multicohort object"""
+    global _multicohort
+    if not _multicohort:
+        _multicohort = create_multicohort()
+    return _multicohort
+
+
+def deprecated_get_cohort() -> Cohort:
     """Return the cohort object"""
     global _cohort
     if not _cohort:
-        _cohort = create_cohort()
+        _cohort = deprecated_create_cohort()
     return _cohort
 
 
-def create_cohort() -> Cohort:
+def get_multicohort() -> Cohort | MultiCohort:
+    """
+    Return the cohort or multicohort object based on the workflow configuration.
+    """
+    input_datasets = config_retrieve(['workflow', 'input_datasets'], [])
+    custom_cohort_ids = config_retrieve(['workflow', 'input_cohorts'], [])
+    if custom_cohort_ids and input_datasets:
+        raise ValueError('Cannot use both custom_cohort_ids and input_datasets in the same workflow')
+
+    # NOTE: When configuring sgs in the config is deprecated, this will be removed.
+    if config_retrieve(['workflow', 'input_cohorts'], []):
+        if not isinstance(custom_cohort_ids, list):
+            raise ValueError('input_cohorts must be a list')
+        return actual_get_multicohort()
+    return deprecated_get_cohort()
+
+
+def create_multicohort() -> MultiCohort:
+    """
+    Add cohorts in the multicohort.
+    """
+    config = config_retrieve(['workflow'])
+    custom_cohort_ids = config_retrieve(['workflow', 'input_cohorts'], [])
+    multicohort = MultiCohort()
+
+    datasets_by_cohort = get_metamist().get_sgs_for_cohorts(custom_cohort_ids)
+
+    read_pedigree = config.get('read_pedigree', True)
+    for cohort_id in custom_cohort_ids:
+        cohort = multicohort.create_cohort(cohort_id)
+        sgs_by_dataset_for_cohort = datasets_by_cohort[cohort_id]
+        populate_cohort(cohort, sgs_by_dataset_for_cohort, read_pedigree=read_pedigree)
+
+    return multicohort
+
+
+def populate_cohort(cohort: Cohort, sgs_by_dataset_for_cohort, read_pedigree: bool = True):
```

Comment:
If this is internal-only like the other `_populate_foo()` functions, maybe mark it as such: `_populate_cohort`?

Reply:
Done :)